sql/semisync_master.h

```diff
@@ -51,6 +51,7 @@ struct Slave_info
 */
   Trans_binlog_info gtid_state_sent;
   Trans_binlog_info gtid_state_ack;
+  bool need_sync;
```

Comment:
`need_sync` doesn't really make sense in the context of `slave_info`; perhaps an `enum` with 3 options for `async`/`semi-sync stalled`/`semi-sync active` would work better, with an added benefit of extensibility.